src/lib/cacheRoutes.ts

```diff
@@ -0,0 +1,56 @@
+// wrapper for worker fetch to cache routes
+// TODO: remove console.logs
+export function cacheRoutes(
+  fetch: (request: Request, env: Env, ctx: ExecutionContext) => Promise<Response>,
+  options: { exclude?: RegExp } = {}
+) {
+  return async (request: Request, env: Env, ctx: ExecutionContext) => {
+    const url = new URL(request.url)
+    let cache: Cache | undefined
+
+    try {
+      // check cache or clear cache when pragma: no-cache
+      if (
+        request.method === 'GET' &&
+        !url.search && // avoid caching for any url with params e.g. &__rsc=true
+        !options.exclude?.test(url.pathname)
+      ) {
+        cache = await caches.open('default')
+        if (request.headers.get('pragma')?.includes('no-cache')) {
+          try {
+            // this exposes cache clearing to the client - not ideal but practical
+            ctx.waitUntil(cache.delete(request))
```

Comment:
@jldec not sure if this helps at all, but just in case: we also provide cloudflare's ExecutionContext (`ctx` over here ^) as `requestInfo.cf`.

Reply:
cool - thanks yeah - that will be helpful if this code moved out of the fetch wrapper into middleware say.